src/core/vectortile/qgsvectortilelayer.cpp

```diff
@@ -385,7 +385,8 @@ bool QgsVectorTileLayer::writeSymbology( QDomNode &node, QDomDocument &doc, QStr
 
 void QgsVectorTileLayer::setTransformContext( const QgsCoordinateTransformContext &transformContext )
 {
-  Q_UNUSED( transformContext )
+  if ( mDataProvider )
+    mDataProvider->setTransformContext( transformContext );
```

Comment:
What happens if the VT layer is invalid initially and only loaded later, won't that miss out on this setter?
I.e. should the context be saved in a member variable of the layer and applied to a dataprovider in loadDataSource() as well?

Reply:
good point!

Reply:
Fixed, with a bunch of related improvements

Reply:
Thanks!